node/actors/consensus/src/replica/state_machine.rs

```diff
@@ -27,47 +24,33 @@ pub(crate) struct StateMachine {
     /// The deadline to receive an input message.
     pub(crate) timeout_deadline: time::Deadline,
     /// A reference to the storage module. We use it to backup the replica state.
-    pub(crate) storage: Arc<dyn ReplicaStateStore>,
+    pub(crate) storage: FallbackReplicaStateStore,
 }
 
 impl StateMachine {
     /// Creates a new StateMachine struct. We try to recover a past state from the storage module,
     /// otherwise we initialize the state machine with whatever head block we have.
     pub(crate) async fn new(
         ctx: &ctx::Ctx,
-        storage: Arc<dyn ReplicaStateStore>,
+        storage: FallbackReplicaStateStore,
```

Comment:
why "fallback"? It is supposed to be the primary source of state, with empty state being the fallback, no?

Reply:
or you mean WithFallback? But then do we need that verbosity?

Reply:
Yeah, I've meant "store with fallback" (fallback being based on a `BlockStore`). Would something like `FullReplicaStateStore` work better?

Reply:
I guess that would be better. But why not just ReplicaStateStore? The fallback is an implementation detail from the pov of this function. Perhaps we should make FallbackReplicaStateStore into a trait instead of ReplicaStateStore?